### PR TITLE
Include default dnf modules in dnf_modules

### PR DESCRIPTION
**Pull request by psegedy**
Currently we are collecting only `enabled` dnf modules but packages can be installed/updated from `active` modules. Active means that module stream is either enabled or default (and not disabled)

- updated  insights-core to v3.1.23 by updating pyproject.toml, requirements.txt and running `poetry lock` with `Poetry (version 1.4.2)` since there is no pin for poetry version
- use DnfModuleList parser to get active (enabled or default) modules

depends on https://github.com/RedHatInsights/insights-core/pull/3756

### CODE REVIEW

File: src/puptoo/process/profile.py [hunk 1]
```diff
@@ -457,18 +457,18 @@ def system_profile(
...
-                for module_name in module.sections():
-                    modules.append(
-                        {
-                            "name": module_name,
-                            "stream": module.get(module_name, "stream"),
-                        }
-                    )
-            profile["dnf_modules"] = sorted(modules, key=lambda k: k["name"])
+            modules = {}
+            for _, module in dnf_module_list.items():
```
Comment (vkrizan) [inline src/puptoo/process/profile.py:463]:
```suggestion
            for module in dnf_module_list.values():
```

File: src/puptoo/process/profile.py [hunk 2]
```diff
@@ -457,18 +457,18 @@ def system_profile(
...
+                if "[e]" in module.stream or ("[d]" in module.stream and "[x]" not in module.stream):
+                    # use only active modules (enabled or default)
+                    stream = module.stream.replace("[d]", "").replace("[e]", "").strip()
```
Comment (vkrizan) [inline src/puptoo/process/profile.py:466]: Maybe cleanup all special bracket parts (to be resilient for future enhancements).

File: src/puptoo/process/profile.py [hunk 3]
```diff
@@ -457,18 +457,18 @@ def system_profile(
...
+                    if "[d]" in module.stream and module.name not in modules:
+                        modules[module.name] = {"name": module.name, "stream": stream}
+                        continue
+                    modules[module.name] = {"name": module.name, "stream": stream}
+            profile["dnf_modules"] = list(sorted(modules.values(), key=lambda k: k["name"]))
```
Comment (vkrizan) [inline src/puptoo/process/profile.py:471]: `list()` call is redundant. `sorted` already returns a list.

**Review by vkrizan — COMMENTED** [review]
Thank you Patrik!

**Comment by psegedy** [comment]
thanks for review @vkrizan, however it will be quite different after changes I had to make to insights-core

**Review by MichaelMraka — APPROVED** [review]
lgtm

**Review by vkrizan — APPROVED** [review]
:heavy_check_mark:

**Comment by psegedy** [comment]
@r14chandra can you take a look at this PR, please?

File: src/puptoo/process/profile.py [hunk 4]
```diff
@@ -475,6 +479,18 @@ def system_profile(
...
+    if dnf_module_list:
+        try:
+            modules = []
+            for name, module in dnf_module_list.items():
+                for stream in module.streams:
+                    if stream.active:
+                        modules.append({"name": name, "stream": stream.stream})
+            profile["dnf_modules"] = sorted(modules, key=lambda k: k["name"])
+        except Exception as e:
+            catch_error("dnf_modules", e)
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:491]:
```suggestion
            catch_error("dnf_module_list", e)
```

**Review by r14chandra — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**